daffodil-test/src/test/scala-new/edu/illinois/ncsa/daffodil/section13/packed/TestPacked.scala

```diff
@@ -54,4 +54,18 @@ class TestPacked {
   // @Test def testHexCharset03(): Unit = { runner.runOneTest("hexCharset03") } // textNumberPattern V symbol
   @Test def testHexCharset04(): Unit = { runner.runOneTest("hexCharset04") }
 
+  @Test def testPackedCharset01(): Unit = { runner.runOneTest("packedCharset01") }
+  @Test def testPackedCharset02(): Unit = { runner.runOneTest("packedCharset02") }
+  @Test def testPackedCharset03(): Unit = { runner.runOneTest("packedCharset03") }
+  //@Test def testPackedCharset04(): Unit = { runner.runOneTest("packedCharset04") } // need packed calendar
+
+  @Test def testBCDCharset01(): Unit = { runner.runOneTest("bcdCharset01") }
+  @Test def testBCDCharset02(): Unit = { runner.runOneTest("bcdCharset02") }
+  @Test def testBCDCharset03(): Unit = { runner.runOneTest("bcdCharset03") }
+  //@Test def testBCDCharset04(): Unit = { runner.runOneTest("bcdCharset04") } // need bcd calendar
+
+  @Test def testIBM4690Charset01(): Unit = { runner.runOneTest("IBM4690Charset01") }
+  @Test def testIBM4690Charset02(): Unit = { runner.runOneTest("IBM4690Charset02") }
+  @Test def testIBM4690Charset03(): Unit = { runner.runOneTest("IBM4690Charset03") }
+  //@Test def testIBM4690Charset04(): Unit = { runner.runOneTest("IBM4690Charset04") } // need bcd calendar
```

Comment:
I'm not sure if this is enough tests to ensure everything is correct. I think we need a lot more tests to make sure the logic in DecimalUtils.scala is correct. I think we also need some tests where we have various lengths (some delimited, some runtime expressions, some fixed), various endianess, and spaning multiple bytes. This is going to need a lot of tests to ensure that everything is correct. There are some tests in the ibm-contributed directory. Some of those might be able to be enabled and be a good starting point.

Reply:
I'd like to see unit tests in daffodil-lib-unitttest for the DecimalUtils.scala library. Basically because it can be tested there. These tests can then be insuring that the prims and parsers are invoking the right decimal util methods based on the properties provided, and detects the right errors, and not so much on whether the underlying library does the basics right.